client/src/raw.rs

```diff
@@ -473,6 +476,21 @@ macro_rules! proxier {
     };
 }
 
+macro_rules! namespaced_request {
+    ($req:ident) => {{
+        let ns_str = $req.get_ref().namespace.clone();
+        // Attach namespace header
+        $req.metadata_mut().insert(
+            TEMPORAL_NAMESPACE_HEADER_KEY,
+            ns_str
+                .parse()
+                .unwrap_or_else(|_| AsciiMetadataValue::from_static("")),
+        );
+        // Init metric labels
+        AttachMetricLabels::namespace(ns_str)
+    }};
+}
```

Comment:
Does this need be a macro?

Reply:
Yes. There's no trait for `.namespace` and adding one would involve writing a whole new procmacro crate. This is the simplest type safe way to do it. That's the same reason there are a lot of macros in general in here, because I have to work with generated code I can't easily control.

Reply:
OK, I was just curious because it seemed structurally similar to the `r.extensions_mut().insert(labels);` call nearby (and wasn't doing anything with symbol information available at compile-time only).